### DISCUSSION TITLE
Apply opacity to events in the past?

### DISCUSSION THREAD

**Discussion by cmjoseph07** [vkurko/calendar#187] [General]
Hello All,

Has anyone figured out a good way to place an opacity to the background of all events in the past?

I am not sure if I should do it at event creation, after mounting, or etc. What would be the best path forward @vkurko?

Thanks!

**Comment by vkurko** [comment]
I think you can use [eventClassNames](https://github.com/vkurko/calendar#eventclassnames) for this. If you provide it as a function then you can test each event for whether it is in the past or not and apply corresponding CSS class for opacity.

**Reply by cmjoseph07** [reply]
@vkurko This worked splendid, thank you for the help! I will close the discussion now.

**Comment by cmjoseph07** [comment]
@vkurko 

One more question, when the nowIndicator passes an event, the background is not updated. I am re-running this function every minute as well.

**Reply by vkurko** [reply]
Yes, the calendar will not automatically redraw events at midnight. To do this, you need to either call `refetchEvents()` or `setOption('event', eventsArray)`, depending on how you pass events.

But the best way, I think, would be if this functionality is implemented inside the calendar, and it itself will set the css class automatically. I'll see what can be done about this.

**Reply by cmjoseph07** [reply]
Okay, I using `refetchEvents()` at the current time and seems to be working. Let me know if you have any update on this!
